### PR TITLE
fix: Added more tests and fixed a couple of issues with product import

### PR DESCRIPTION
**Pull request by sradevski**
The git tree has deepened a bit, but I expect to merge all of these soon (so please review 😄 )
CLOSES CC-249

### CODE REVIEW

File: packages/core/core-flows/src/product/workflows/import-products.ts [hunk 1]
```diff
@@ -30,6 +30,23 @@ export const importProductsWorkflow = createWorkflow(
 
     waitConfirmationProductImportStep()
 
+    // Q: Can we somehow access the error from the step that threw here? Or in a compensate step at least?
```
Comment (olivermrbl) [inline packages/core/core-flows/src/product/workflows/import-products.ts:33]: cc @carlos-r-l-rodrigues

File: packages/core/core-flows/src/product/helpers/normalize-for-import.ts [hunk 2]
```diff
@@ -148,10 +151,19 @@ const normalizeVariantForImport = (
           { currency_code: priceKey.toLowerCase(), amount: normalizedValue },
         ]
       } else {
+        const region = regionsMap.get(priceKey)
+        if (!region) {
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Region with ID ${priceKey} not found`
+          )
+        }
```
Comment (olivermrbl) [inline packages/core/core-flows/src/product/helpers/normalize-for-import.ts:160]: question: Do we want to change this to region name now, or would you rather wait for a follow-up PR?

File: packages/core/core-flows/src/notification/steps/notify-on-failure.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+export const notifyOnFailureStepId = "notify-on-failure"
+export const notifyOnFailureStep = createStep(
+  notifyOnFailureStepId,
+  async (data: NotifyOnFailureStepInput) => {
+    return new StepResponse(void 0, data)
+  },
+
+  async (data, { container }) => {
+    console.log("HERE!!!")
```
Comment (olivermrbl) [inline packages/core/core-flows/src/notification/steps/notify-on-failure.ts]: todo: The good ol' 😄 
```suggestion
```

Comment (sradevski) [reply packages/core/core-flows/src/notification/steps/notify-on-failure.ts]: Ahaha classic 🤦 we should probably add a console.log linter...

File: packages/core/core-flows/src/product/helpers/normalize-for-import.ts (hunk 2 above)
Comment (sradevski) [reply packages/core/core-flows/src/product/helpers/normalize-for-import.ts:160]: I already created a separate ticket, so in a follow up PR

File: packages/core/core-flows/src/product/workflows/import-products.ts (hunk 1 above)
Comment (sradevski) [reply packages/core/core-flows/src/product/workflows/import-products.ts:33]: Merged it but this is still an open question (created a Linear ticket to track it)

Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/product/workflows/import-products.ts:33]: I'm not sure I follow here what is needed here. Something can fail but the workflow will continue to be executed?

At the moment using this workflows api we are not exposing anything to access other steps responses/errors.
[Here is the context](https://github.com/medusajs/medusa/blob/24c105f28837a1190da35da1335b6b3cedc8eed0/packages/core/workflows-sdk/src/utils/composer/create-step.ts#L130-L143) we are proving to the step, but main invoke argument `stepArguments` has everything.

@adrien2p @olivermrbl I'm fine with exposing more properties in this context, like the transaction and the invoke and compensation responses. At least people could find their way if needed. Or we think about exposing compensation responses, transaction errors or transaction/step states/status.

Comment (adrien2p) [reply packages/core/core-flows/src/product/workflows/import-products.ts:33]: we could eventually wrap it under another getter, something like `getErrorFromStep(stepName)` and so on if we want to make it easy for them to discover and find those kind of information?

**Pull request merged**